Update README.md

katiejohnso opened the pull request:
The shared Google Doc was used extensively for the last half of the questions

Thank you!!!!!!!!!!!!

katiejohnso committed:
Update README.md

Jvonderhbuff commented:
Hey Katie!

Not sure how to make the in-code comments, so will make mine in a list here with the line number noting where my comment is from.

16: I am not sure if this will work for reflexive. For an object property to be reflexive, the relationship in question must always apply to itself. I can certainly see the cases you mention where a specific relationship would be reflexive, but I am not sure that the same could be said for all has_quality relationships. For example, we might say github has_quality confusing, but this does not mean that github has_quality github in the same sense. You might be fine to reuse your symmetric and reflexive relationship for this.

45: I just want to say that this seems like a really good example.

49: Not sure why you have this line if your answer is on a separate line.

50: Not sure about the explanation here. I think the relationship is good, but your explanation is a bit confusing as you appear to argue against your conclusion there.

68: I think you need underscores at a couple of points here. I am pretty sure the relationships should be marked with underscores, and you may want to have underscores to mark which qualities are one thing (i.e. has_arms_on_tuesday vs. has_arms on tuesday)

Also you may need to add the definitions for each concept/part of the formulation here. For example, here is what I have for this:
Tuesday is instance_of one-dimensional temporal region 
Wednesday is an instance of one-dimensional temporal region 
Arm is an instance of fiat object part
“Sally participates in having at least one arm on Tuesday” is an instance of occurrent 
“Sally participates in having no arms on Wednesday” is an instance of occurrent
“Sally participates in having at least one arm on Tuesday precedes Sally participates in having no arms on Wednesday”

77: It is worth mentioning that you could also model this as Goofus and Gallant not being married to each other.

86: This looks good to me. The only thing worth mentioning is that you could also do this by having "has arms on Tuesday" and "has arms on Wednesday" as single properties to simplify this a bit.

124: Worth noting that they need not be married to each other here. The phrasing is ambiguous. For my own thoughts on this question, you could probably have a third relationship to convey that the marriage is between two people. You might try a trinary relationship to say "x is married to y at t" but I am instinctively uneasy at placing people and a time in the same relationship.

131: If that is the picture I think it is, I think it is incomplete. You might be better off just stating the relationships rather than trying to do a full graph like that (unless you are feeling confident with mermaid). I have included what I did to give you an idea of what you might do for this:
Content is an instance_of generic dependent continuant
File is an instance_of generic dependent continuant
File 1 Turtle implements content 1
File 2 XML implements content 1
Files 1 and 2 are output of Computer 1
Computer 1 is bearer of File 1 and File 2
Computer 1 is an instance_of material entity
Computer 1 sends output to Computers 2 and 3
File 1 is input of Computer 2
Computer 2 receives input from Computer 1
Computer 2 is an instance_of material entity
File 2 is input of Computer 3
Computer 3 receives input from Computer 1
Computer 3 is an instance_of material entity
Computer 1 2 and 3 are part of Network 1
Network 1 is an instance_of Network
Network is_a object aggregate

172: You might be better off including a brief explanation of your answer here.

175: I think you need to adjust the example here. You would be better off just saying "committee" instead of "subcommittee," as one could think that a subcommittee is different from a committee.

201: I think it would be good to include some lines breaking down the categories and relationships here. The graph explains the model but having the parts is also good to have.

205-206: You might want to add a line about when we wouldn't want to use Pellet. It may not be necessary but it was asked for from me, at least, for the explanations in project 2.

Those are my comments. Everything else looks good to me. Glad to give you a review and hope this won't take too long to fix up!

katiejohnso commented:
Thank you so much for these comments!

16:  I agree with you.  The definition of has_quality is "a relation between an independent continuant (the bearer) and a quality, in which the quality specifically depends on the bearer for its existence." I've argued that this is a reflexive relation because, by the provided definition, the quality specifically depends on the bearer for its existence.  My original thought was this: if the quality "is confusing" depends on the bearer of that quality, in this case, Github, to exist, then maybe this is a reflexive relation.  That seems wrong now.

45: THANK YOU!

49 +50: I like to include as much reasoning in my answers as possible when I'm not really sure what I'm doing.  The horseshoe crabs and armadillos information is an attempt to think through whether or not "has skeleton" could be understood on non-asymmetric terms. The only two object properties that are labeled as asymmetric in BFO are "has skeleton" and "has role in modeling."  I found it difficult to argue why these should not be labeled as asymmetric properties (as asked by the prompt).  Overall I'm unconvinced these should not be labeled as asymmetric, but I tried a couple ideas out (which is why they conflict and feel somewhat watery). 

68: I agree and added in some underscores (though I'm not sure if I added them in correctly).  I also added in the suggested descriptions.

124: I agree with you.  Defining marriage is confusing and frustrating to me (in class and in this project, apparently).  I'm still not sure what to make of it in this question.

131:  For the sake of learning I want to leave this graph in, even if it is incomplete or incorrect.  It will be easier for me to understand why and how it is incomplete or incorrect in graph form, rather than in statements.  That being said, thank you for including this option in your comment.  This is super helpful to look at outside of the graph.  

172:  You're right.  I looked back at BFO in Protege and am more confused now, but it was interesting to read the definition of "immaterial entity" and the relationship between the immaterial and material. 

175:  But couldn't using "committee" rather than "subcommittee" require even further explanation about how the second committee is a part of the first?  My thinking here is that a subcommittee is an independent committee, but one that is a part or section of the original committee. 

201:  Thank you for this suggestion

205-206: Added this in, thank you!
